### ISSUE TITLE
Upgrading from v7 to v8, chat messages are not shown or logged in console

### ISSUE THREAD

**Issue by c0wg0d** [kangarko/ChatControl-Pro#1343]
Instead of using our channel format in the console for server chat, all we see is the following:

[23:32:43 INFO]: Sending 'hiya'
[23:32:45 INFO]: Sending 'hey'
[23:32:51 INFO]: Sending 'what's up'
[23:32:56 INFO]: Sending 'nothing'
[23:32:59 INFO]: Sending 'oh lol'
[23:33:35 INFO]: Sending 'me too'
[23:33:52 INFO]: Sending 'want to go mining'
[23:33:59 INFO]: Sending 'sure'

There should be player names, server/channel prefix, etc.

Using ChatControlPro 8.3.6.  I also have DiscordSRV installed if that makes a difference.

One of our example channel configs from formatting.yml:

```
  testchannel:
    Bracket:
      Message: '&7[&1T&7]&f '
    Player:
      Message: '{pl_prefix}{nick}{pl_suffix}&7: &r'
      Hover_Event:
        Action: SHOW_TEXT
        Values:
        - '&7Name&f: {player} {javascript_is_helper}{javascript_is_seniorhelper}{javascript_is_moderator}{javascript_is_admin}{javascript_is_owner}
          {javascript_is_vip}'
        - '&7Time Played&f: {statistic_time_played}'
        - '&9Click to PM&f {player}'
      Click_Event:
        Action: SUGGEST_COMMAND
        Value: '/pm {player} '
    Msg:
      Message: '{message}'
```

**Comment by kangarko** [comment]
Please use the latest version.

So you are using channels from channels.yml and when an user types a message, nothing is shown in the console besides the "Sending ..." message?

As far as I tested, after removing that debug message, the console receives channel output properly.

![a](https://i.imgur.com/HDWQYAE.png)

**Comment by c0wg0d** [comment]
I might know what the problem is.  Can you please tell me what ```Bungee_Server_Name``` is for in settings.yml?  This appears to be a new setting with ChatControl v8.

**Comment by kangarko** [comment]
Here's the documentation:
![a](https://i.imgur.com/uVnBDBW.png)

**Comment by c0wg0d** [comment]
I was hoping for a little more insight.  The fact is, this is a new setting, and cross-server communication works just fine without it.  I currently am using a single settings.yml file symlinked across all my Paper/Spigot instances and it works just fine, but now with this apparently new setting that is somehow required, I need to duplicate the settings.yml file across all my servers just for this single setting?  Why?  And why does not having this setting cause the chat in the current server to not be saved to the log file when all the other cross server chat IS logged just fine?  I am really dumbfounded as to why my previous config worked just fine with v7, but now with v8 is doesn't, when this is seemingly the only thing that really changed for cross-server chat and logging.

**Comment by kangarko** [comment]
Due to recent changes in Spigot, we can no longer identify the server by reaching for the server.properties file like we used to.

As such, to link servers properly by their names we added an explicit 'Bungee_Server_name' identifier to settings.yml that is used to match the server name from your BungeeCord config.yml to correctly sync data between servers.

If you're symlinking your settings.yml then you're overriding this option which will break the plugin.

Worst case, I can move that setting out to a separate file, let me know if that's an option.

**Comment by c0wg0d** [comment]
I have many Spigot servers, and maintaining a separate settings file for each would be extremely cumbersome when I want to change a non-Bungee setting.  I currently have ALL YML files symlinked across all my Spigot servers.  The only things I don't symlink are the logs folder, the data.db file, and the error.log file.

If you could move all Bungee related settings that need to be unique per server into a bungee.yml file, that would be much easier management-wise for anyone who runs multiple servers like me.  Thank you.

**Comment by c0wg0d** [comment]
I'm still not sure why messages are not being logged to the console on the server where the message originated from.  I have been looking over my config files in depth for hours.  I'm getting quite frustrated.  What else should I be looking for?  Should I post all my configs?

**Comment by c0wg0d** [comment]
I am adding the zip file containing all my config files as you suggested in the pinned issue.

[ChatControl.zip](https://github.com/kangarko/ChatControl-Pro/files/3535664/ChatControl.zip)

**Comment by kangarko** [comment]
> I have many Spigot servers, and maintaining a separate settings file for each would be extremely cumbersome when I want to change a non-Bungee setting. For example, if I wanted to add a new death message, I would have to copy that change to every single config file across all my servers. Not my idea of a good time. I currently have ALL YML files symlinked across all my Spigot servers. The only things I don't symlink are the logs folder, the data.db file, and the error.log file.
> 
> If you could move all Bungee related settings that need to be unique per server into a bungee.yml file, that would be much easier management-wise for anyone who runs multiple servers like me. Thank you.

I will do it, sounds great! Is it sufficient if there's just the "bungee-server-name" option in the bungee.yml file as you proposed? Or do you need more bungee related options moved there as well?





> I have updated to the latest ChatControlPro version 8.3.12. I have explicitly specified a Console_Format setting in channels.yml. I have separated settings.yml for every Paper/Spigot instance and added the corresponding Bungee_Server_Name setting for each.
> 
> I'm still not sure why messages are not being logged to the console on the server where the message originated from. When the debug message was there, it was at least logging SOMETHING, now that you have removed that it doesn't print anything at all. I have been looking over my config files in depth for hours. I looked through every setting with a fine toothed comb and even rewrote my config files from scratch using the latest yml files in the 8.3.12 jar file. I'm getting quite frustrated. What else should I be looking for? Should I post all my configs? This still seems like a straight up bug to me, not intended behavior.

It's not a bug, it's a feature :P Since you had Cancel_Event on true, we cancel your chat event and cancelled events won't fire a console message.

I added a condition that if you have Console_Log we log the message explicitly when the cancel event flag is false.

**Comment by c0wg0d** [comment]
Wow, I totally missed that.  I had Cancel_Event set to true because of this bug in the past: https://github.com/kangarko/ChatControl-Pro/issues/194

I set it to false and it both fixed the issue (I see chat in the console now!) and also it doesn't appear to have those issues with Discord anymore.  PROBLEM SOLVED!  Thank you so much.

> Is it sufficient if there's just the "bungee-server-name" option in the bungee.yml file as you proposed? Or do you need more bungee related options moved there as well?

The two settings I had to change in settings.yml were Server_Name and Bungee_Server_Name.  If those two settings could go in a separate bungee.yml file, then I would be able to share the rest of the settings.yml file across all my servers, making it possible to symlink them all to a single file.  This would be a great addition.  Thank you for offering this solution!
